### PR TITLE
interface

### LINKED ISSUE #54: InterfaceStoneAgeGame

### LINKED ISSUE #55: InterfaceStoneAgeObservable

### LINKED ISSUE #56: InterfaceStoneAgeObserver

### LINKED ISSUE #57: InterfaceGetState

### LINKED ISSUE #59: InterfaceFeedTribe

### LINKED ISSUE #60: InterfaceNewTurn

### LINKED ISSUE #61: InterfacePlayerBoardGameBoard

### LINKED ISSUE #62: CivilisationCard datatype

### LINKED ISSUE #63: ImmediateEffect

### LINKED ISSUE #64: EndOfGameEffect

### LINKED ISSUE #65: Player datatype

### LINKED ISSUE #66: InterfaceFigureLocationInternal

### LINKED ISSUE #67: EvaluateCivilisationCardImmediateEffect inteface

### LINKED ISSUE #68: InterfaceFigureLocation

### LINKED ISSUE #69: InterfaceToolUse

### LINKED ISSUE #70: InterfaceTakeReward

### PR DESCRIPTION
**Pull request by johankogoodvibes**
resolves #54 
resolves #55
resolves #56  
resolves #57 
resolves #59   
resolves #60   
resolves #61
resolves #62     
resolves #63 
resolves #64 
resolves #65  
resolves #66  
resolves #67 
resolves #68 
resolves #69 
resolves #70

### CODE REVIEW

File: src/main/java/sk/uniba/fmph/dcs/game_board/Player.java
Comment (Duchyna1) [inline src/main/java/sk/uniba/fmph/dcs/game_board/Player.java:1]: toto musime nejako zistit, mozno sa ho spytat

File: src/main/java/sk/uniba/fmph/dcs/stone_age/CivilisationCard.java
Comment (Duchyna1) [inline src/main/java/sk/uniba/fmph/dcs/stone_age/CivilisationCard.java:1]: aj toto

File: src/main/java/sk/uniba/fmph/dcs/game_board/Player.java
Comment (johankogoodvibes) [reply src/main/java/sk/uniba/fmph/dcs/game_board/Player.java:1]: No ja som všetko dal do toho priečinku lebo tam boli nejaké tie enumy a veci co on nakodil a boli už hotové v tom gameboard priečinku ak to myslis

File: src/main/java/sk/uniba/fmph/dcs/game_board/Player.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package sk.uniba.fmph.dcs.game_board;
+
+import sk.uniba.fmph.dcs.stone_age.InterfacePlayerBoardGameBoard;
+import sk.uniba.fmph.dcs.stone_age.PlayerOrder;
+
+/**
+ * @param playerOrder Player's order in the game
+ * @param playerBoard Interface for the player's board in the game
+ */
+public record Player(PlayerOrder playerOrder, InterfacePlayerBoardGameBoard playerBoard) { }
```
Comment (Duchyna1) [inline src/main/java/sk/uniba/fmph/dcs/game_board/Player.java:10]: treba asi nieco nakodit

File: src/main/java/sk/uniba/fmph/dcs/stone_age/CivilisationCard.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package sk.uniba.fmph.dcs.stone_age;
+
+public record CivilisationCard(ImmediateEffect[] immediateEffect, EndOfGameEffect[] endOfGameEffect) { }
```
Comment (Duchyna1) [inline src/main/java/sk/uniba/fmph/dcs/stone_age/CivilisationCard.java:3]: aj tu

**Review by Duchyna1 — CHANGES_REQUESTED** [review]

File: src/main/java/sk/uniba/fmph/dcs/stone_age/CivilisationCard.java (hunk 2 above)
Comment (johankogoodvibes) [reply src/main/java/sk/uniba/fmph/dcs/stone_age/CivilisationCard.java:3]: No nečítal som si pravidlá ale ak to správne chápem tak ten record ma všetky užitočné veci by default v compilatore, teda constructor, gettery,... čiže ak niečo z pravidiel nehovorí ze to ma obsahovať nejake funkcie, alebo tam niečo nenapísal tak je to vpoho podla mňa

Comment (Duchyna1) [inline src/main/java/sk/uniba/fmph/dcs/stone_age/CivilisationCard.java:3]:
```suggestion
// TODO
// public record CivilisationCard(ImmediateEffect[] immediateEffect, EndOfGameEffect[] endOfGameEffect) { }
```

**Review by Duchyna1 — CHANGES_REQUESTED** [review]

**Review by Duchyna1 — APPROVED** [review]

**Pull request merged**